### DISCUSSION TITLE
Introduce a transfer list component

### DISCUSSION THREAD

**Discussion by naveen2593kumar** [ing-bank/lion#1149] [Open for contributions]
Hey,

**Just a bit about me:**
 I am Naveen Kumar, working with ING Bank. Using ing-web for FE application and component implementation for DARE project.

**Now coming to the Idea:**
I have this idea to implement a component of transfer list (like this https://material-ui.com/components/transfer-list/)
this is a great to have widget in any UI component set.

Also I wanna work on it with you guys.
I have a decent exposure in ing-web/lion-web..
I am starting the draft implementation on this, soon I will try to raise a pull request.

**Comment by jorenbroekema** [comment]
Aaah yeah that's a cool component. We actually have a listbox now that supports multiple choice, so I guess this component could be made by combining two multiple choice listboxes with some buttons in the middle that make the transfer happen.

There is a nice WAI ARIA example of this pattern https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/listbox/listbox.html , especially example 2 which has multiple choice. Should be doable now that we have listbox 👍 

I think it would be cool to have lion-transfer with a multiple-choice option (allow to toggle it) with those 4 buttons as a start in lion, and have the subclasser extend it

@tlouisse  you have any ideas about it?

**Comment by tlouisse** [comment]
Cool idea @naveen2593kumar 

I agree indeed with @jorenbroekema. These are listboxes and there is a very similar WAI ARIA pattern implementing this, so we should take this as a lead 👍 
You might also want to try implementing this, since it kind of fits in here: https://github.com/ing-bank/lion/discussions/977 (I should have a wip branch somewhere if you would be interested in that).

lion-transfer is indeed something we also can consider, but maybe first focus on building a docs example purely with two listboxes, so we can think of the proper abstractions once this example is in place?

**Comment by naveen2593kumar** [comment]
Great to hear from you, even I was thinking in the same way. Using 2 listbox (in multiple-choice mode) in middle 4 buttons 
shiftAllToRight
shiftSelectedToRight
shiftSelectedToLeft
shiftAllToLeft

I will start the implementation and if it seems fine, I will raise a pull request...

**Reply by naveen2593kumar** [reply]
Old PR closed , please refer new PR : https://github.com/ing-bank/lion/pull/1232
